app/services/code_climate/projects_summary_service.rb

```diff
@@ -38,12 +39,21 @@ def open_issues_count_average
     end
 
     def ratings
-      code_climate_ratings = code_climate_metrics.pluck(:code_climate_rate).compact
-      code_climate_ratings.each_with_object(Hash.new(0)) do |code_climate_rate, ratings|
-        ratings[code_climate_rate] += 1
+      @code_climate_ratings = code_climate_metrics.pluck(:id,
+                                                         :code_climate_rate,
+                                                         'code_climate_project_metrics.created_at')
+      @code_climate_ratings = sort_ratings_by_date
+      @code_climate_ratings.each_with_object(Hash.new(0)) do |code_climate_metric, ratings|
+        letter = code_climate_metric[1]
+        ratings[letter] += 1 if letter.present?
       end
     end
 
+    def sort_ratings_by_date
+      @code_climate_ratings.sort_by { |id_rate_date| [id_rate_date[0], id_rate_date[2]] }
+                           .each_with_object({}) { |id_rate, hsh| hsh[id_rate[0]] = id_rate[1] }
```

Comment:
what is this doing? the method is called `sort_...` but the sort was already done in the previous line

Reply:
I want to keep only the last rate per project, so after the sorting by date, I store the last value for each project.